packages/shared-integration/src/defaults.ts

```diff
@@ -15,8 +15,8 @@ export const defaultFilesystemGlobs = [
 export const defaultIdeMatchInclude: RegExp[] = [
   // String literals
   /(['"`])[^\1]*?\1/g,
-  // HTML tags ignore [>]
-  /<(?:[^<>\[\]]+|\[[^\]]*\])*>/g,
+  // HTML tags
+  /<[^\/](?:[^<>\[\]"]+)(?:>|"[^"]*"(?:(?:[^<>\[\]{]+)(?:>|{[^}]*}|(?:\[[^\]]*\])))*|(?:\[[^\]]*\](?:(?:[^<>\[\]{"]+)(?:>|"[^"]*"|{[^}]*}|(?:\[[^\]]*\])))*))/g,
```

Comment:
This seems to be over complicated to me

Maybe:

```suggestion
  /<(?:"[^"]*"|'[^"]*'|[^>])+>/g,
```

Reply:
<img width="675" alt="image" src="https://github.com/unocss/unocss/assets/57086651/3744b816-f227-41e5-b80c-5a51bb89b1ab">
He will miss some needed matches